### PR TITLE
VMR-1680: Add icon property to ExternalCommand class

### PR DESCRIPTION
**Pull request by nmontmarquette**
The Shotgun Create application needed a quick and easy way to access a command's icon.

### CODE REVIEW

**Comment by nmontmarquette** [comment]
I'm using the serialized ExternalCommand object from within the ` view-master/src/ui/QtoolkitDriver.cpp` to package an extra icon field to the UI (React) code.

File: python/external_config/external_command.py [hunk 1]
```diff
@@ -419,6 +432,7 @@ def _execute(self, pre_cache=False, entity_ids=None):
                 supports_multiple_selection=self._sg_supports_multiple_selection,
                 pre_cache=pre_cache,
                 pythonpath=current_pypath,
+                # Todo: Do I need to add icon here as well?
```
Comment (nmontmarquette) [inline python/external_config/external_command.py]: Also, I wasn't sure about that one

Comment (manneohrstrom) [reply python/external_config/external_command.py]: no need to pass the icon here!

Comment (nmontmarquette) [reply python/external_config/external_command.py]: Good I'll clean that one line up then

**Comment by nmontmarquette** [comment]
Do you require me to add test? I believe this file has none.

File: python/external_config/external_command.py [hunk 2]
```diff
@@ -237,6 +243,13 @@ def display_name(self):
         """
         return self._display_name
 
+    @property
+    def icon(self):
+        """
+        The path to a square png icon file representing this item
```
Comment (manneohrstrom) [inline python/external_config/external_command.py]: we should add : "or None if no icon is associated"

Comment (nmontmarquette) [reply python/external_config/external_command.py]: Should the property be called icon_256 ? for consistency with overall TK interface ?

Comment (nmontmarquette) [reply python/external_config/external_command.py]: Not sure whether it's None, I really just copy over what I get from the `data` dict, I would assume it might be empty string

Comment (thebeeland) [reply python/external_config/external_command.py]: I'd prefer to stick with "icon" instead of "icon_256"

Comment (thebeeland) [reply python/external_config/external_command.py]: My guess would be a `None` and not an empty string, but I'm not 100% sure on that.

File: tests/external_config/__init__.py [hunk 3]
```diff
@@ -7,3 +7,91 @@
...
 # agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
 # not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import os
+
+from mock import Mock
+
+# import the test base class
+test_python_path = os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "python"))
```
Comment (jfboismenu) [inline tests/external_config/__init__.py]: No need to add to sys.path for. The toolkit unit test framework does it for you.

File: tests/external_config/test_external_command.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    @property
+    def ec_data(self):
+        return self._data
+
+    @property
+    def ec(self):
+        return self._external_command
+
+    def test_properties(self):
+        self.assertEquals(self.ec.pipeline_configuration_name, self._pipeline_configuration_id)
```
Comment (jfboismenu) [inline tests/external_config/test_external_command.py]: Missing docstring.

File: tests/external_config/test_external_command.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        self.assertEquals(self.ec.is_group_default, self.ec_data["group_default"])
+        if self.ec_data["sg_deny_permissions"]:
+            self.assertEquals(self.ec.excluded_permission_groups_hint, self.ec_data["sg_deny_permissions"])
+        else:
+            self.assertEquals(self.ec.excluded_permission_groups_hint, [])
+        self.assertEquals(self.ec.support_shotgun_multiple_selection, self.ec_data["sg_supports_multiple_selection"])
+        self.assertEquals(self.ec.tooltip, self.ec_data["tooltip"])
+        self.assertTrue(len(repr(self.ec)) > 0)
+
+    def test_serialize_deserialize(self):
```
Comment (jfboismenu) [inline tests/external_config/test_external_command.py]: Missing docstring.

File: tests/external_config/test_external_command.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        # Creating the external command object
+        self._external_command = self.external_config.ExternalCommand.create(
+            self.external_config_loader,
+            self.ec_data,
+            "Task"
+        )
+
+    @property
+    def ec_data(self):
+        return self._data
```
Comment (jfboismenu) [inline tests/external_config/test_external_command.py]: Missing docstring.

File: tests/external_config/test_external_command.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+    @property
+    def ec(self):
```
Comment (jfboismenu) [inline tests/external_config/test_external_command.py]: Missing docstring.
